docs/authoring.md

```diff
@@ -162,12 +159,33 @@ Consuming C#/WinRT components from MSIX-packaged applications is supported with
 
 C#/WinRT supports authoring out-of-process components that can be consumed by Windows Runtime compatible languages. Currently, consuming an out-of-process component is supported for managed C# apps with the use of a packaging project. This scenario requires creating a `WinRT.Host.runtimeconfig.json` file, as demonstrated in the [background task component sample](https://github.com/microsoft/CsWinRT/tree/master/src/Samples/BgTaskComponent). Native C++ consumption of out-of-process C# components is not fully supported yet.
 
+### Using Windows App SDK
+
+Authoring WinUI types with C#/WinRT is in development and is supported with some exceptions.
+Component authors can implement WinUI types using the Windows App SDK class library template. 
+Components are consumed by projects made from the various other Windows App SDK templates.
+
+When using a project reference, projects made from the following templates are not supported currently: C++ app with WAP, C++ Unpackaged, and C# Unpackaged. 
+
+Using a package reference to the authored component will work unless authoring a user control, which is not supported yet.
+
+When authoring a custom user control, consuming the component via NuGet package is not supported currently.
+
+If a C++ app is consuming a component that provides a custom user control, an additional entry needs to be made in the `pch.h` header file and in the manifest. 
+
+For the `pch.h` file, add `#include <winrt/MyAuthoredComponent.MyAuthoredComponent_XamlTypeInfo.h>`.
+For the manifest, add an additional `ActivatableClass` entry for `MyAuthoredComponent.MyAuthoredComponent_XamlTypeInfo.XamlMetaDataProvider`.

```

Comment:
Probably add the code snippet here so folks can copy/paste?

Reply:
You can link to the existing instructions here: https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/troubleshooting#symptoms-and-remedies
See the last entry in that list

Reply:
@asklar From what I see, this is needed for activating XAML types; do you have additional info on why we need this?

Reply:
as far as the xamltype info header, this is necessary for things like XAML `Binding` to work; @RealTommyKlein can probably say better words :D
the activatable class entry is necessary to tell winrt "this is the library that knows how to create this type"